tests/flows/test_flow.py

```diff
@@ -26,17 +36,170 @@ def check_flow(flow):
+    def test_get_flow(self):
+        # We need to use the production server here because 4024 is not the test
+        # server
+        openml.config.server = self.production_server
+
+        flow = openml.flows.get_flow(4024)
+        self.assertIsInstance(flow, openml.OpenMLFlow)
+        self.assertEqual(flow.flow_id, 4024)
+        self.assertEqual(len(flow.parameters), 24)
+        self.assertEqual(len(flow.components), 1)
+
+        subflow_1 = list(flow.components.values())[0]
+        self.assertIsInstance(subflow_1, openml.OpenMLFlow)
+        self.assertEqual(subflow_1.flow_id, 4025)
+        self.assertEqual(len(subflow_1.parameters), 14)
+        self.assertEqual(subflow_1.parameters['E'], 'CC')
+        self.assertEqual(len(subflow_1.components), 1)
+
+        subflow_2 = list(subflow_1.components.values())[0]
+        self.assertIsInstance(subflow_2, openml.OpenMLFlow)
+        self.assertEqual(subflow_2.flow_id, 4026)
+        self.assertEqual(len(subflow_2.parameters), 13)
+        self.assertEqual(subflow_2.parameters['I'], '10')
+        self.assertEqual(len(subflow_2.components), 1)
+
+        subflow_3 = list(subflow_2.components.values())[0]
+        self.assertIsInstance(subflow_3, openml.OpenMLFlow)
+        self.assertEqual(subflow_3.flow_id, 1724)
+        self.assertEqual(len(subflow_3.parameters), 11)
+        self.assertEqual(subflow_3.parameters['L'], '-1')
+        self.assertEqual(len(subflow_3.components), 0)
+
+    def test_from_xml_to_xml(self):
+        # Get the raw xml thing
+        # TODO maybe get this via get_flow(), which would have to be refactored to allow getting only the xml dictionary
+        for flow_id in [1185, 1244, 1196, 1112, ]:
+            flow_xml = _perform_api_call("flow/%d" % flow_id)[1]
+            flow_dict = xmltodict.parse(flow_xml)
+
+            flow = openml.OpenMLFlow._from_xml(flow_dict)
+            new_xml = flow._to_xml()
 
+            flow_xml = flow_xml.replace('  ', '').replace('\t', '').strip().replace('\n\n', '\n').replace('&quot;', '"')
+            flow_xml = re.sub(r'^$', '', flow_xml)
+            new_xml = new_xml.replace('  ', '').replace('\t', '').strip().replace('\n\n', '\n').replace('&quot;', '"')
+            new_xml = re.sub(r'^$', '', new_xml)
+
+            self.assertEqual(new_xml, flow_xml)
+
+    def test_to_xml_from_xml(self):
+        scaler = sklearn.preprocessing.StandardScaler(with_mean=False)
+        boosting = sklearn.ensemble.AdaBoostClassifier(
+            base_estimator=sklearn.tree.DecisionTreeClassifier())
+        model = sklearn.pipeline.Pipeline(steps=(
+            ('scaler', scaler), ('boosting', boosting)))
+        flow = openml.flows.sklearn_to_flow(model)
+        flow.flow_id = -234
+        # end of setup
+
+        xml = flow._to_xml()
+        xml_dict = xmltodict.parse(xml)
+        new_flow = openml.flows.OpenMLFlow._from_xml(xml_dict)
```

Comment:
This shows that the function ``_from_xml`` should be called ``_from_dict``. Currently, this is a round-trip:
``openml.flows.OpenMLFlow._from_xml(asdf._to_dict())``. That's weird naming ;)